oauthdoctor/oauth/helper_test.go

```diff
@@ -3,9 +3,163 @@ package oauth
 import (
 	"io/ioutil"
 	"log"
+	"os"
 	"testing"
+
+	"github.com/googleads/google-ads-doctor/oauthdoctor/diag"
 )
 
+type FakeConfig struct {
+	cfgFile diag.ConfigFile
+}
+
+func (c *FakeConfig) ReplaceConfig(k, v string) string {
+	c.cfgFile.SetConfigKeys(k, v)
+	return ""
+}
+
+func TestReplaceCloudCredentials(t *testing.T) {
+	log.SetOutput(ioutil.Discard)
+
+	originalReadID := readClientID
+	originalReadSecret := readClientSecret
+	defer func() {
+		readClientID = originalReadID
+		readClientSecret = originalReadSecret
+	}()
+
+	test := struct {
+		desc         string
+		c            FakeConfig
+		clientID     string
+		clientSecret string
+	}{
+		desc: "Successful Replacement",
+		c: FakeConfig{
+			cfgFile: diag.ConfigFile{
+				ConfigKeys: diag.ConfigKeys{
+					ClientID:     "oldID",
+					ClientSecret: "oldSecret",
+				},
+			},
+		},
+		clientID:     "newID",
+		clientSecret: "newSecret",
+	}
+
+	readClientID = func() string {
+		return test.clientID
+	}
+	readClientSecret = func() string {
+		return test.clientSecret
+	}
+
+	replaceCloudCredentials(&test.c)
+
+	if test.c.cfgFile.ClientID != test.clientID || test.c.cfgFile.ClientSecret != test.clientSecret {
+		t.Errorf("[%s] got: (ClientID=%s, ClientSecret=%s), want: (ClientID=%s, ClientSecret=%s)",
+			test.desc, test.c.cfgFile.ClientID, test.c.cfgFile.ClientSecret, test.clientID, test.clientSecret)
+	}
+}
+
+func setup() func() {
```

Comment:
Consider renaming this function to be more explicit about the setup that is being done. setup() is usually the name of a procedure called in TestMain().